### PR TITLE
[ARTIC-1903] fix Minor Optimizing scans files with iceberg sequence and add more test cases for hive support table

### PR DESCRIPTION
**Pull request by wangtaohz**
## Why are the changes needed?

hot-fix for #1093 

## Brief change log

  - add planFilesWithSequence for ChangeTableIncrementalScan
  - add task property `move-files-to-hive-location` for hive support major optimize to avoid move files to hive location unexpectedly
  - fix hive support major optimize keeping optimizing with 1 small file
  - fix expireTaskHistory NPE
  - refactor OptimizingIntegrationTest and add more test case for hive support table
  - remove misuse of org.glassfish.jersey.internal.guava.*

## How was this patch tested?
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

## Documentation

  - Does this pull request introduces a new feature? (yes / **no**)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / **not documented**)

### CODE REVIEW

File: ams/ams-server/src/main/java/com/netease/arctic/ams/server/optimize/AbstractArcticOptimizePlan.java [hunk 1]
```diff
@@ -208,38 +212,37 @@ protected void addChangeFilesIntoFileTree() {
...
+      markChangeSequence(partition, file.getSequenceNumber());
 
       addCnt.getAndIncrement();
     });
     LOG.debug("{} ==== {} add {} change files into tree, total files: {}." + " After added, partition cnt of tree: {}",
         tableId(), getOptimizeType(), addCnt, unOptimizedChangeFiles.size(), partitionFileTree.size());
   }
 
-  private long getMaxTransactionIdLimit(List<PrimaryKeyedFile> unOptimizedChangeFiles) {
+  private long getMaxSequenceLimit(List<ContentFileWithSequence<?>> unOptimizedChangeFiles) {
```
Comment (zhoujinsong) [inline ams/ams-server/src/main/java/com/netease/arctic/ams/server/optimize/AbstractArcticOptimizePlan.java:245]: Minor optimize change files with the order of the change file sequence will result in changes files with larger transaction Id are merged before changes files with smaller transaction Id when streaming and batch writing occur simultaneously.

How should we solve this problem?

File: ams/ams-server/src/main/java/com/netease/arctic/ams/server/optimize/BasicOptimizeCommit.java [hunk 2]
```diff
@@ -228,6 +228,7 @@ private void minorCommit(ArcticTable arcticTable,
       AtomicInteger addedPosDeleteFile = new AtomicInteger(0);
       StructLikeMap<Long> partitionOptimizedSequence =
           TablePropertyUtil.getPartitionOptimizedSequence(arcticTable.asKeyedTable());
+      AtomicBoolean shouldCommit = new AtomicBoolean();
```
Comment (zhoujinsong) [inline ams/ams-server/src/main/java/com/netease/arctic/ams/server/optimize/BasicOptimizeCommit.java:231]: Maybe we can support `skipEmptyCommit` method in `OverwriteBaseFiles`.

Comment (wangtaohz) [reply ams/ams-server/src/main/java/com/netease/arctic/ams/server/optimize/BasicOptimizeCommit.java:231]: It seems to be more reasonable. I will change it.

**Review by zhoujinsong — APPROVED** [review]
LGTM.

**Pull request merged**